telemetry/metrics_test.go

```diff
@@ -22,6 +22,15 @@ func TestMetricPayload(t *testing.T) {
 		Timestamp:  now,
 		Value:      1.0,
 	})
+	h.RecordMetric(Summary{
+		Name:       "another-metric",
+		Attributes: map[string]interface{}{"zip": "zap"},
+		Timestamp:  now,
+		Count:      4.0,
+		Sum:        1.0,
+		Min:        math.NaN(),
+		Max:        3.0,
```

Comment:
Probably worth adding a test to check if `Max` correctly marshals `math.NaN()` as well.